src/scheduled_tasks.rs

```diff
@@ -450,6 +452,62 @@ async fn update_banned_when_expired(pool: &mut DbPool<'_>) {
   }
 }
 
+/// Find all unpublished posts with scheduled date in the future, and publish them.
+async fn publish_scheduled_posts(context: &Data<LemmyContext>) {
+  let pool = &mut context.pool();
+  let conn = get_conn(pool).await;
+
+  match conn {
+    Ok(mut conn) => {
+      let scheduled_posts: Vec<_> = post::table
+        .inner_join(community::table)
+        .inner_join(person::table)
+        // find all posts which have scheduled_publish_time that is in the  past
+        .filter(post::scheduled_publish_time.is_not_null())
+        .filter(coalesce(post::scheduled_publish_time, now()).lt(now()))
+        // make sure the post, person and community are still around
+        .filter(not(post::deleted.or(post::removed)))
+        .filter(not(person::banned.or(person::deleted)))
+        .filter(not(community::removed.or(community::deleted)))
+        // ensure that user isnt banned from community
+        .filter(not(exists(
+          community_person_ban::table
+            .filter(community_person_ban::community_id.eq(community::id))
+            .filter(community_person_ban::person_id.eq(person::id)),
+        )))
+        .select((post::all_columns, community::all_columns))
+        .get_results::<(Post, Community)>(&mut conn)
+        .await
+        .inspect_err(|e| error!("Failed to read unpublished posts: {e}"))
+        .ok()
+        .unwrap_or_default();
+
+      for (post, community) in scheduled_posts {
```

Comment:
This fetching and looping isn't necessary, see my last comment on this about moving it into impls/post

Reply:
Im not sure how you mean. See my [other reply](https://github.com/LemmyNet/lemmy/pull/5025#discussion_r1763183868).

Reply:
Mmmk you have to send the federation and webmention anyway, so a loop is required anyway. Also tho, you should be able to do `diesel::update(filtered_posts).set(....` , but I wouldn't worry about it since you have to loop anyway for federation.